src/Command/Resources/ResourcesSetCommand.php

```diff
@@ -232,18 +238,28 @@ protected function execute(InputInterface $input, OutputInterface $output)
             // Set the instance count.
             if (!$service instanceof Service) { // a Service instance count cannot be changed
                 if (isset($givenCounts[$name])) {
-                    $instanceCount = $givenCounts[$name];
-                    if ($instanceCount !== $properties['instance_count'] && !($instanceCount === 1 && !isset($properties['instance_count']))) {
-                        $updates[$group][$name]['instance_count'] = $instanceCount;
+                    if (!empty($autoscalingEnabled[$name])) {
+                        $this->stdErr->writeln(sprintf('The instance count of the service <error>%s</error> cannot be changed when autoscaling is enabled.', $name));
+                        $this->stdErr->writeln('');
+                        $errored = true;
+                    } else {
+                        $instanceCount = $givenCounts[$name];
+                        if ($instanceCount !== $properties['instance_count'] && !($instanceCount === 1 && !isset($properties['instance_count']))) {
+                            $updates[$group][$name]['instance_count'] = $instanceCount;
+                        }
                     }
                 } elseif ($showCompleteForm) {
                     $ensureHeader();
-                    $default = $properties['instance_count'] ?: 1;
-                    $instanceCount = $questionHelper->askInput('Enter the number of instances', $default, [], function ($v) use ($name, $service, $instanceLimit) {
-                        return $this->validateInstanceCount($v, $name, $service, $instanceLimit);
-                    });
-                    if ($instanceCount !== $properties['instance_count']) {
-                        $updates[$group][$name]['instance_count'] = $instanceCount;
+                    if (!empty($autoscalingEnabled[$name])) {
+                        $this->stdErr->writeln(sprintf('The instance count of the service <error>%s</error> cannot be changed when autoscaling is enabled.', $name));
```

Comment:
I think we don't need this error, we can skip showing the instance count question entirely if autoscaling is enabled. Proposing a refactor in: https://github.com/platformsh/legacy-cli/pull/1556